dotenv/parse.ts

```diff
@@ -27,30 +27,29 @@ function expandCharacters(str: string): string {
     ($1: keyof typeof CHARACTERS_MAP): string => CHARACTERS_MAP[$1] ?? "",
   );
 }
-function expand(str: string, variablesMap: { [key: string]: string }): string {
-  if (EXPAND_VALUE_REGEXP.test(str)) {
-    return expand(
-      str.replace(EXPAND_VALUE_REGEXP, function (...params) {
-        const {
-          inBrackets,
-          inBracketsDefault,
-          notInBrackets,
-          notInBracketsDefault,
-        } = params[params.length - 1];
-        const expandValue = inBrackets || notInBrackets;
-        const defaultValue = inBracketsDefault || notInBracketsDefault;
-
-        let value: string | undefined = variablesMap[expandValue];
-        if (value === undefined) {
-          value = Deno.env.get(expandValue);
-        }
-        return value === undefined ? expand(defaultValue, variablesMap) : value;
-      }),
-      variablesMap,
-    );
-  } else {
-    return str;
+
+function expand(str: string, variablesMap: Record<string, string>): string {
+  let current = str;
+
+  while (EXPAND_VALUE_REGEXP.test(current)) {
+    current = current.replace(EXPAND_VALUE_REGEXP, (...params) => {
+      const {
+        inBrackets,
+        inBracketsDefault,
+        notInBrackets,
+        notInBracketsDefault,
+      } = params.at(-1);
+
+      const expandValue = inBrackets ?? notInBrackets;
+      const defaultValue = inBracketsDefault ?? notInBracketsDefault;
+
+      return (
+        variablesMap[expandValue] ?? Deno.env.get(expandValue) ?? defaultValue
```

Comment:
The existing code seems calling `expand` on `defaultValue`?
```suggestion
        variablesMap[expandValue] ?? Deno.env.get(expandValue) ?? expand(defaultValue, variablesMap)
```